framework/web/UrlRule.php

```diff
@@ -216,6 +221,145 @@ public function init()
     }
 
     /**
+     * Initializes normalizer configuration
+     * @param UrlManager $manager the URL manager
+     * @throws \Exception if config is not understood
+     */
+    protected function initNormalizer($manager) {
+        $normalize = $this->normalize;
+
+        $configNoAction = [
+            // whether to collapse multiple slashes to single
+            'collapse-slashes' => false,
+            // remove trailing slash if it exists
+            'remove-trailing-slash' => false,
+            // add trailing slash if there is no one
+            'add-trailing-slash' => false,
+            // what action to perform if pathInfo was changed during normalization
+            // redirect: instant redirect to normalized path
+            // route: return new route (must be configured via 'route')
+            // none: no action, just route by normalized path
+            'action' => 'none',
+            // string: route for $action == 'route', i.e. 'site/redirect'
+            'route' => null
+        ];
+
+        if (is_string($normalize) || is_null($normalize)) {
+            switch ($normalize) {
+                case '':
+                case 'safe-remove-trailing-slash':
```

Comment:
Do you mean that instead

```
        $configNoAction = [
            'collapse-slashes' => false,
            'trailing-slash' => false,
            'action' => 'none',
            'route' => null
        ];
```

is better to write something like:

```
        $configNoAction = [
            self::NORMALIZE_COLLAPSE_SLASHES => false,
            self::NORMALIZE_TRAILING_SLASH => false,
            self::NORMALIZE_ACTION => self::NORMALIZE_ACTION_NONE,
            self::NORALIZE_ROUTE => null
        ];
```

like this?

Second example looks ugly to me.. I've never seen such practice in ```yii-config.php```.